package/base-files/Makefile

```diff
@@ -96,6 +96,10 @@ define ImageConfigOptions
 	echo 'pi_preinit_net_messages="$(CONFIG_TARGET_PREINIT_SHOW_NETMSG)"' >>$(1)/lib/preinit/00_preinit.conf
 	echo 'pi_preinit_no_failsafe_netmsg="$(CONFIG_TARGET_PREINIT_SUPPRESS_FAILSAFE_NETMSG)"' >>$(1)/lib/preinit/00_preinit.conf
 	echo 'pi_preinit_no_failsafe="$(CONFIG_TARGET_PREINIT_DISABLE_FAILSAFE)"' >>$(1)/lib/preinit/00_preinit.conf
+ifndef CONFIG_TARGET_PREINIT_USER_CONFIG_REMOVED
+	echo 'pi_user_config_wait_timeout="$(if $(CONFIG_TARGET_PREINIT_USER_CONFIG_TIMEOUT),$(CONFIG_TARGET_PREINIT_USER_CONFIG_TIMEOUT),4)"' >>$(1)/lib/preinit/00_preinit.conf
+	echo 'pi_no_user_config="$(CONFIG_TARGET_PREINIT_USER_CONFIG_DISABLED)"' >>$(1)/lib/preinit/00_preinit.conf
+endif
```

Comment:
Some notes for if you get bored... user + config is too generic for my liking... possibly something like "buttonmenu" or "preinit-custom" for a more generic flavour